File: CPP/Arrays/selectionSort.cpp
#include <iostream>

using namespace std;

int main() {
  int n;
  cin>>n;

  int arr[n];

  for(int i =0; i < n; i++) {
    cin>>arr[i];
  }
  // loop through the array and find the next minimum element
  for (int i = 0; i < n-1; i++) {
    for (int j = i + 1; j < n; j++){
      // find the next minimum element and swap it with i 
      if (arr[j] < arr[i]) {
        int temp = arr[j];
        arr[j] = arr[i];
        arr[i] = temp;
      }
    }
  }

  for(int i = 0; i < n; i++) {
    cout<<arr[i]<<" ";
  }
}

File: sorting/selectionSort.js

function selectionSort(arr) {
  for (let i = 0; i < arr.length; i++) {
    let minIdx = i;
    for (let j = i; j < arr.length; j++) {
      if (arr[j] < arr[minIdx]) {
        minIdx = j;
      }
    }
    swap(arr, minIdx, i);
  }
  return arr;
}


function swap(arr, idx1, idx2) {
  let temp = arr[idx1];
  arr[idx1] = arr[idx2];
  arr[idx2] = temp;
}

console.log(selectionSort([2,4,5,6,3,2,1,1,0]));